lib/Command/AddCommand.php

```diff
@@ -0,0 +1,170 @@
+			->setDescription('Add a new guest account')
+			->addArgument(
+				'created-by',
+				InputArgument::REQUIRED,
+				'User ID who is set as creator'
+			)
+			->addArgument(
+				'uid',
+				InputArgument::REQUIRED,
+				'User ID used to login (must only contain a-z, A-Z, 0-9, -, _ and @)'
+			)
+			->addArgument(
+				'email',
+				InputArgument::REQUIRED,
+				'Email address'
+			)
+			->addOption(
+				'generate-password',
+				null,
+				InputOption::VALUE_NONE,
+				'Do not set an initial password'
+			)
+			->addOption(
+				'password-from-env',
+				null,
+				InputOption::VALUE_NONE,
+				'Read password from environment variable OC_PASS'
+			)
+			->addOption(
+				'display-name',
+				null,
+				InputOption::VALUE_OPTIONAL,
+				'User name used in the web UI (can contain any characters)',
+				''
+			)
+			->addOption(
+				'language',
+				null,
+				InputOption::VALUE_OPTIONAL,
+				'Language',
+				''
+			);
+		parent::configure();
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		$creatorUser = $this->userManager->get($input->getArgument('created-by'));
+		if ($creatorUser === null) {
+			$output->writeln('<error>The user "' . $input->getArgument('created-by') . '" does not exist.</error>');
+			return 1;
+		}
+
+		$uid = $input->getArgument('uid');
+		if ($this->userManager->userExists($uid)) {
+			$output->writeln('<error>The user "' . $uid . '" already exists.</error>');
+			return 1;
+		}
+
+		$email = $input->getArgument('email');
+		if (!$this->mailer->validateMailAddress($email)) {
+			$output->writeln('<error>Invalid email address "' . $email . '".</error>');
+			return 1;
+		}
+
+		$password = null;
+		if (!$input->getOption('generate-password')) {
+			if ($input->getOption('password-from-env')) {
+				$password = getenv('OC_PASS');
+				if (!$password) {
+					$output->writeln('<error>--password-from-env given, but OC_PASS is empty!</error>');
+					return 1;
+				}
+			} elseif ($input->isInteractive()) {
+				/** @var QuestionHelper $helper */
+				$helper = $this->getHelper('question');
+
+				$question = new Question('Enter password: ');
+				$question->setHidden(true);
+				$password = $helper->ask($input, $output, $question);
+
+				$question = new Question('Confirm password: ');
+				$question->setHidden(true);
+				$confirm = $helper->ask($input, $output,$question);
+
+				if ($password !== $confirm) {
+					$output->writeln("<error>Passwords did not match!</error>");
+					return 1;
+				}
+			} else {
+				$output->writeln("<error>Interactive input or --password-from-env is needed for entering a password!</error>");
+				return 1;
+			}
+		}
+
+		try {
+			$user = $this->guestManager->createGuest(
+				$creatorUser,
+				$input->getArgument('uid'),
+				$email,
+				$input->getOption('display-name') ?? '',
+				$input->getOption('language') ?? '',
+				$password
+			);
+		} catch (\Exception $e) {
+			$output->writeln('<error>' . $e->getMessage() . '</error>');
```

Comment:
the problem with this is that it hides the trace. and tracing back an error just from the exception message isn't always possible. You could try the ``\Throwable::getTraceAsString`` and write that to the output maybe?

Reply:
I copy-pasted this from a command in the server (occ user:add), I'll look up another less old instance which might already have such approach

Reply:
or maybe just not catch the exception.

I checked all other commands I could find and most don't bother catching exceptions, which will likely result in them being printed to the console anyway

Reply:
meh... not catching the exception also automatically doesn't print the trace:
<img width="365" alt="image" src="https://user-images.githubusercontent.com/277525/102456214-1ab42e80-4041-11eb-8896-eb4b44dc7268.png">

Reply:
okay, after a quick Google it seems that one needs to add `-v` to any command to see traces when they are caught by Symfony:
```
% occ guests:add admin guest9 guest1@example.com -v
Enter password: 
Confirm password: 

In GuestManager.php line 101:
                  
  [Exception]     
  Test exception  
                  

Exception trace:
  at /srv/www/htdocs/server/apps/guests/lib/GuestManager.php:101
 OCA\Guests\GuestManager->createGuest() at /srv/www/htdocs/server/apps/guests/lib/Command/AddCommand.php:150
 OCA\Guests\Command\AddCommand->execute() at /srv/www/htdocs/server/3rdparty/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at /srv/www/htdocs/server/3rdparty/symfony/console/Application.php:1009
 Symfony\Component\Console\Application->doRunCommand() at /srv/www/htdocs/server/3rdparty/symfony/console/Application.php:273
 Symfony\Component\Console\Application->doRun() at /srv/www/htdocs/server/3rdparty/symfony/console/Application.php:149
 Symfony\Component\Console\Application->run() at /srv/www/htdocs/server/lib/private/Console/Application.php:215
 OC\Console\Application->run() at /srv/www/htdocs/server/console.php:99
 require_once() at /srv/www/htdocs/server/occ:11
```

so I'll remove the catch block